cluster-up/cluster/k8s-provider-common.sh

```diff
@@ -9,28 +9,45 @@ source "${KUBEVIRTCI_PATH}/cluster/ephemeral-provider-common.sh"
 #if UNLIMITEDSWAP is set to true - Kubernetes workloads can use as much swap memory as they request, up to the system limit.
 #otherwise Kubernetes workloads can use as much swap memory as they request, up to the system limit by default
 function configure_swap_memory () {
-  if [ "$KUBEVIRT_SWAP_ON" == "true" ] ;then
-
-    for nodeNum in $(seq -f "%02g" 1 $KUBEVIRT_NUM_NODES); do
-        if [ ! -z $KUBEVIRT_SWAP_SIZE_IN_GB  ]; then
-          $ssh node${nodeNum} -- sudo dd if=/dev/zero of=/swapfile count=$KUBEVIRT_SWAP_SIZE_IN_GB bs=1G
-          $ssh node${nodeNum} -- sudo mkswap /swapfile
-        fi
-
-        $ssh node${nodeNum} -- sudo swapon -a
-
-        if [ ! -z $KUBEVIRT_SWAPPINESS ]; then
-          $ssh node${nodeNum} -- "sudo /bin/su -c \"echo vm.swappiness = $KUBEVIRT_SWAPPINESS >> /etc/sysctl.conf\""
-          $ssh node${nodeNum} -- sudo sysctl vm.swappiness=$KUBEVIRT_SWAPPINESS
-        fi
+    if [ "$KUBEVIRT_SWAP_ON" == "true" ] ;then
+      for nodeNum in $(seq -f "%02g" 1 $KUBEVIRT_NUM_NODES); do
+          if [ ! -z $KUBEVIRT_SWAP_SIZE_IN_GB  ]; then
+            $ssh node${nodeNum} -- sudo dd if=/dev/zero of=/swapfile count=$KUBEVIRT_SWAP_SIZE_IN_GB bs=1G
+            $ssh node${nodeNum} -- sudo mkswap /swapfile
+          fi
+
+          $ssh node${nodeNum} -- sudo swapon -a
+
+          if [ ! -z $KUBEVIRT_SWAPPINESS ]; then
+            $ssh node${nodeNum} -- "sudo /bin/su -c \"echo vm.swappiness = $KUBEVIRT_SWAPPINESS >> /etc/sysctl.conf\""
+            $ssh node${nodeNum} -- sudo sysctl vm.swappiness=$KUBEVIRT_SWAPPINESS
+          fi
+
+          if [ $KUBEVIRT_UNLIMITEDSWAP == "true" ]; then
+            $ssh node${nodeNum} -- "sudo sed -i ':a;N;\$!ba;s/memorySwap: {}/memorySwap:\n  swapBehavior: UnlimitedSwap/g'  /var/lib/kubelet/config.yaml"
+            $ssh node${nodeNum} -- sudo systemctl restart kubelet
+          fi
+      done
+    fi
+}
 
-        if [ $KUBEVIRT_UNLIMITEDSWAP == "true" ]; then
-          $ssh node${nodeNum} -- "sudo sed -i ':a;N;\$!ba;s/memorySwap: {}/memorySwap:\n  swapBehavior: UnlimitedSwap/g'  /var/lib/kubelet/config.yaml"
-          $ssh node${nodeNum} -- sudo systemctl restart kubelet
-        fi
-  done
-fi
+function configure_ksm_module () {
+    if [ "$KUBEVIRT_KSM_ON" == "true" ] ;then
+      for nodeNum in $(seq -f "%02g" 1 $KUBEVIRT_NUM_NODES); do
+        $ssh node${nodeNum} -- "sudo echo 1 | sudo tee /sys/kernel/mm/ksm/run >/dev/null"
```

Comment:
And also, you don't need the first sudo
```suggestion
        $ssh node${nodeNum} -- "echo 1 | sudo tee /sys/kernel/mm/ksm/run >/dev/null"
```

Reply:
Actually without the `sudo` i'm getting 
```
tee: /sys/kernel/mm/ksm/pages_to_scan: Permission denied
make: *** [Makefile:143: cluster-up] Error 1
```